src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -422,6 +443,46 @@ public IDisposable SuspendChangeNotifications(SuspensionMode mode)
         }
 
         /// <summary>
+        /// Synchroniz the list with the suspenstion context.
+        /// </summary>
+        private void SynchronizeFromSuspensionContext()
+        {
+            if (_suspensionContext.NewItems.Count > 0)
+            {
+                _suspensionContext.NewItemIndices.Reverse();
```

Comment:
Instead of reversing twice and iteration forward, just iterate backwards. So you safe the reversing.